Core/src/main/java/com/plotsquared/core/util/task/TaskManager.java

```diff
@@ -25,144 +25,269 @@
 import lombok.Getter;
 import lombok.Setter;
 
+import javax.annotation.Nonnegative;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
 import java.util.Collection;
 import java.util.HashMap;
 import java.util.HashSet;
 import java.util.Iterator;
-import java.util.concurrent.atomic.AtomicBoolean;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Callable;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
 import java.util.concurrent.atomic.AtomicInteger;
 
+/**
+ * Task manager that handles scheduling of tasks.
+ * Synchronous methods make no guarantee of being scheduled on the
+ * server thread, instead they guarantee that no two synchronous
+ * operations happen at the same time. Implementations of
+ * the task manager might make other guarantees. All asynchronous
+ * operations will happen on another thread, no matter where
+ * they're scheduled from.
+ */
 public abstract class TaskManager {
 
-    public static final HashSet<String> TELEPORT_QUEUE = new HashSet<>();
-    public static final HashMap<Integer, Integer> tasks = new HashMap<>();
+    private static final Set<String> teleportQueue = new HashSet<>();
+    private static final Map<Integer, PlotSquaredTask> tasks = new HashMap<>();
+
     public static AtomicInteger index = new AtomicInteger(0);
 
-    @Getter @Setter private static TaskManager implementation;
+    @Getter @Setter private static TaskManager platformImplementation;
 
-    public static int runTaskRepeat(Runnable runnable, int interval) {
-        if (runnable != null) {
-            if (getImplementation() == null) {
-                throw new IllegalArgumentException("disabled");
-            }
-            return getImplementation().taskRepeat(runnable, interval);
-        }
-        return -1;
+    /**
+     * Add a string to the teleport queue
+     *
+     * @param string String to add
+     */
+    public static void addToTeleportQueue(@Nonnull final String string) {
+        teleportQueue.add(string);
     }
 
-    public static int runTaskRepeatAsync(Runnable runnable, int interval) {
+    /**
+     * Remove a string from the teleport queue
+     *
+     * @param string String to remove
+     * return {@code true} if the value was stored in the map, or {@code false}
+     */
+    public static boolean removeFromTeleportQueue(@Nonnull final String string) {
+        return teleportQueue.remove(string);
+    }
+
+    /**
+     * Add a task to the task map
+     *
+     * @param task Task
+     * @param id   Task ID
+     */
+    public static void addTask(@Nonnull final PlotSquaredTask task, final int id) {
+        tasks.put(id, task);
+    }
+
+    /**
+     * Remove a task from the task map and return the stored value
+     *
+     * @param id Task ID
+     * @return Task if stored, or {@code null}
+     */
+    @Nullable public static PlotSquaredTask removeTask(final int id) {
+        return tasks.get(id);
```

Comment:
This only gets but doesn't remove the task

Reply:
whoops lol